src/qt/transactionrecord.cpp

```diff
@@ -163,8 +164,31 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
                 if (ExtractDestination(txout.scriptPubKey, address))
                 {
                     // Sent to Bitcoin Address
-                    sub.type = TransactionRecord::SendToAddress;
-                    sub.address = CBitcoinAddress(address).ToString();
+										sub.address = CBitcoinAddress(address).ToString();
+										// Sent to testnet or mainnet vote / beacon address
+										if (sub.address == "mk1e432zWKH1MW57ragKywuXaWAtHy1AHZ" || sub.address == "S67nL4vELWwdDVzjgtEP4MxryarTZ9a8GB")
+										{
+												std::vector<std::pair<std::string, std::string>> vTxNormalInfoIn = GetTxNormalBoincHashInfo(wtx);
+												QString TxMessage = MakeSafeMessage(vTxNormalInfoIn[1].second).c_str();
```

Comment:
r u on slack? i suggest a rewrite of this area. its doing twice the work and it can be done better and will gladly help u with this.